File: DesignPatternsInAutomatedTesting-Series/PatternsInAutomation.Tests/Advanced/BingMainPage/BingMainPage.cs
﻿using OpenQA.Selenium;
using PatternsInAutomatedTests.Advanced.Core;

namespace PatternsInAutomatedTests.Advanced.BingMainPage
{
    public class BingMainPage : BasePage<BingMainPageElementMap, BingMainPageValidator>
    {
        public BingMainPage()
           : base(@"http://www.bing.com/")
        {
        }

        public void Search(string textToType)
        {
            this.Map.SearchBox.Clear();
            this.Map.SearchBox.SendKeys(textToType);
            this.Map.GoButton.Click();
        }
    }
}

File: DesignPatternsInAutomatedTestingJava-Series/src/test/java/pageobjectadvanced/pages/bingmainpage/BingMainPage.java
package pageobjectadvanced.pages.bingmainpage;

import pageobjectadvanced.core.BasePage;

public class BingMainPage extends BasePage<BingMainPageElements, BingMainPageAssertions> {
    public BingMainPage() {
        super("http://www.bing.com/");
    }

    public void search(String textToType) {
        elements().searchBox().clear();
        elements().searchBox().sendKeys(textToType);
        elements().goButton().click();
    }
}
